spec/amber/cli/commands/generator_spec.cr

```diff
@@ -16,13 +16,88 @@ module Amber::CLI
           Amber::CLI::Spec.cleanup
         end
-      end
 
-      context "controllers" do
-        it "generates controller with correct verbs and actions" do
+        it "follows naming conventions for all files and class names" do
           ENV["AMBER_ENV"] = "test"
-          MainCommand.run ["new", TESTING_APP]
+          Amber::CLI::MainCommand.run ["new", TESTING_APP]
           Dir.cd(TESTING_APP)
+
+          camel_case = "PostComment"
+          snake_case = "post_comment"
+          class_definition_prefix = "class #{camel_case}"
+          spec_definition_prefix = "describe #{camel_case}"
+
+          [camel_case, snake_case].each do |arg|
+            MainCommand.run ["generate", "scaffold", arg, "name:string"]
+
+            File.exists?("./spec/models/#{snake_case}_spec.cr").should be_true
+            File.exists?("./src/models/#{snake_case}.cr").should be_true
+            File.exists?("./spec/controllers/#{snake_case}_controller_spec.cr").should be_true
+            File.exists?("./src/controllers/#{snake_case}_controller.cr").should be_true
+            File.exists?("./src/views/#{snake_case}/_form.slang").should be_true
+            File.exists?("./src/views/#{snake_case}/edit.slang").should be_true
+            File.exists?("./src/views/#{snake_case}/index.slang").should be_true
+            File.exists?("./src/views/#{snake_case}/new.slang").should be_true
+            File.exists?("./src/views/#{snake_case}/show.slang").should be_true
+
+            File.read("./spec/models/#{snake_case}_spec.cr").should contain spec_definition_prefix
+            File.read("./src/models/#{snake_case}.cr").should contain class_definition_prefix
+            File.read("./spec/controllers/#{snake_case}_controller_spec.cr").should contain spec_definition_prefix
+            File.read("./src/controllers/#{snake_case}_controller.cr").should contain class_definition_prefix
+            File.read("./src/views/#{snake_case}/_form.slang").should contain snake_case
+            File.read("./src/views/#{snake_case}/edit.slang").should contain camel_case
+            File.read("./src/views/#{snake_case}/index.slang").should contain camel_case
+            File.read("./src/views/#{snake_case}/new.slang").should contain camel_case
+            File.read("./src/views/#{snake_case}/show.slang").should contain snake_case
+
+            File.delete("./spec/models/#{snake_case}_spec.cr")
+            File.delete("./src/models/#{snake_case}.cr")
+            File.delete("./spec/controllers/#{snake_case}_controller_spec.cr")
+            File.delete("./src/controllers/#{snake_case}_controller.cr")
+            File.delete("./src/views/#{snake_case}/_form.slang")
+            File.delete("./src/views/#{snake_case}/edit.slang")
+            File.delete("./src/views/#{snake_case}/index.slang")
+            File.delete("./src/views/#{snake_case}/new.slang")
+            File.delete("./src/views/#{snake_case}/show.slang")
```

Comment:
If one expectation above fails these files will not be deleted. Or presence of this files are guaranteed by `Amber::CLI::Spec.cleanup` ? If not, you may need to use `try ... ensure` block for this:

```crystal
try
  # expectations
ensure
  # delete files
end
```

Reply:
The calls to `Amber::CLI::Spec.cleanup` should ensure cleanup, but admittedly, I didn't explicitly test for or look for it.

I did run a bunch of failing test suites though, and there were never any leftovers (according to git).

The entire spec in this file is wrapped in a `begin ... ensure ... end` that executes `Amber::CLI::Spec.cleanup` which is more comprehensive than my file deletion.

I can rerun some failing tests and explicitly look for leftover files (as they may be gitignored).

Reply:
I made it fail one of the `should` tests on generating a scaffold, then did an `ls tmp` and it was empty. It looks like we are good to count on `Amber::CLI::Spec.cleanup`

Reply:
So you don't need to delete it manually, right ?

Reply:
When the test suite is complete, `Amber::CLI::Spec.cleanup` makes sure everything is cleaned up.

The test itself tests two cases of possible CLI input (in a loop) - `PostComment` and `post_comment`

The reason for the manual delete is to avoid false green tests with the second iteration of the loop.

Reply:
Ah, sorry, i haven't noticed the loop above. Thanks for explanation!